src/Microsoft.ManagedZLib/src/IHuffmanTree.cs

```diff
@@ -288,7 +288,7 @@ private void CreateTable()
             uint index = bitBuffer & (uint)_tableMask;
             // If it's negative and it's not in the dictionary, 
             // process the symbol
-            if (!_symDict.ContainsValue(symbol))
+            if (!_symDict.ContainsKey(index))
```

Comment:
Consider using `TryGetValue` instead of `ContainsKey` to avoid doing a double-lookup:
```suggestion
            if (!_symDict.TryGetValue(index, out symbol))
```